analysis/src/CompletionFrontEnd.ml

```diff
@@ -777,6 +778,12 @@ let completionWithParser1 ~currentFile ~debug ~offset ~path ~posCursor ~text =
           (Pos.toString posCursor) (Pos.toString posNoWhite)
           (Loc.toString expr.pexp_loc)
     in
+    (match findThisExprLoc with
+    | Some loc when expr.pexp_loc = loc -> (
+      match exprToContextPath expr with
+      | None -> ()
+      | Some ctxPath -> setResult (Cpath ctxPath))
+    | _ -> ());
```

Comment:
This is pretty messy to include like this in the existing autocomplete. But, it's a pretty valuable feature, and this part of the code base needs a big refactor soon anyway, so I think it's OK given what it enables.